src/App.tsx

```diff
@@ -1,15 +1,27 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState<string | null>(null);
+  const vissibleMovies = query
+    ? moviesFromServer.filter(movie => (
+      movie.title.toLowerCase().includes(query?.trim().toLowerCase())
```

Comment:
split this with variables
Also, DRY, move this check if `str` includes `substr` to the separate function

Reply:
Done